utilities/transactions/optimistic_transaction_test.cc

```diff
@@ -308,6 +308,44 @@ TEST_F(OptimisticTransactionTest, FlushTest2) {
   delete txn;
 }
 
+TEST_F(OptimisticTransactionTest, FlushTest3) {
```

Comment:
The PR offers this as the purpose of the test: "add a test to make sure older snapshot doesn't filter a history memtable covering it." Firstly it would be good to document the purpose of the test here as well. Secondly, I cannot see immediately if the test does what it says. The default number of memtables is 2 so after only a flush we would have two memtables both of them active (nothing left for the history).
Perhaps it would be more clear if i) we make max_write_buffer_number and max_write_buffer_number_to_maintain explicit in the test. ii) flush multiple times until the memtable containing the conflicting key is not present in memory anymore. iii) after each flush check if the conflict is detected correctly.

Reply:
After a flush succeeds, it is removed from immutable list. If it is within max_write_buffer_number_to_maintain, then it is in the history memtable list. I didn't explicitly set max_write_buffer_number_to_maintain, which may be a bug. I'll add it, and do what you suggested, do some more checking to validate the internal behavior.